app/plugin_event.go

```diff
@@ -0,0 +1,20 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+package app
+
+import (
+	"github.com/mattermost/mattermost-server/model"
+)
+
+// notifyClusterPluginEvent publishes `event` to other clusters.
+func (a *App) notifyClusterPluginEvent(event string, data model.PluginEventData) {
+	if a.Cluster != nil {
+		a.Cluster.SendClusterMessage(&model.ClusterMessage{
+			Event:            event,
+			SendType:         model.CLUSTER_SEND_RELIABLE,
+			WaitForAllToSend: true,
```

Comment:
Is there any downside to setting this to be `true`?

Reply:
If notifying other clusters takes longer than "expected" the rest call may time out. Could just turn this into async by setting this to `false`. 0/5 on this.

Reply:
Looks like we use this in `InvalidateAllCaches`, which might otherwise lead to a correctness issue. Stray plugins / missing plugins would be similarly problematic, so 👍 for `WaitForAllToSend`.